src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023holostiminterface.py

```diff
@@ -0,0 +1,242 @@
+                self.image_metadata["SI.hRoiManager.scanFrameRate"]
+            ),  # Frame rate on the entire volumetric stack
+            description="Imaging plane for the holographic stimulation",
+            device=device,
+            excitation_lambda=600.0,  # TODO update
+            indicator="GFP",  # TODO update
+            location="V1",  # TODO update
+            grid_spacing=fov_size_in_um / frame_dimesion,
+            grid_spacing_unit="micrometers",
+            origin_coords=self.rois_metadata["imagingRoiGroup"]["rois"]["scanfields"]["centerXY"],
+            origin_coords_unit="micrometers",
+        )
+        targeted_plane_segmentation = PlaneSegmentation(
+            name=self.targeted_plane_segmentation_name,
+            description="Targeted ROIs from the ScanImage metadata",
+            imaging_plane=imaging_plane,
+        )
+
+        # Set space to pixel transorm matrix
+        pixel_to_space_matrix = self.rois_metadata["imagingRoiGroup"]["rois"]["scanfields"]["pixelToRefTransform"]
+        space_to_pixel_matrix = np.linalg.inv(pixel_to_space_matrix)
+        for roi in self.rois_metadata["integrationRoiGroup"]["rois"]:
+            X = roi["scanfields"]["centerXY"][0]
+            Y = roi["scanfields"]["centerXY"][1]
+            Z = 1
+            spatial_coordinates_roi_centroid = np.array([X, Y, Z])
+            pixel_coordinates_roi_centroid = space_to_pixel_matrix @ spatial_coordinates_roi_centroid
+            pixel_coordinates_roi_centroid = np.round(pixel_coordinates_roi_centroid).astype(int)
+            targeted_plane_segmentation.add_roi(pixel_mask=[pixel_coordinates_roi_centroid])
+
+        nwbfile.processing["ophys"]["ImageSegmentation"].add_plane_segmentation(targeted_plane_segmentation)
+
+        stimulus_table = PatternedOptogeneticStimulusTable(
+            name="PatternedOptogeneticStimulusTable", description="Patterned stimulus"
+        )
+        for trial, hologram_index in enumerate(self._trial_to_stimulation_ids_map):
+            # 7expt has incomplete data
+            if trial >= self._total_number_of_trials:
+                break
+            # create hologram from hologram list
+            if hologram_index != 0:  #
+                hologram_index = hologram_index - 1
+                roi_group = self._hologram_list[hologram_index]
+                roi_indexes = roi_group[~np.isnan(roi_group)].astype(int)
+                if len(roi_indexes) > 0:
+                    # targeted_rois = targeted_rois[roi_group] # se puntassero a indice
+                    targeted_rois = targeted_plane_segmentation.create_roi_table_region(
+                        name="targeted_rois",
+                        description="targeted rois",
+                        region=[
+                            i
+                            for i, roi_index in enumerate(self._targeted_to_segmented_roi_ids_map)
+                            if roi_index in roi_indexes
+                        ],
```

Comment:
I think this part would be more readable if you assign a variable to it first, but keep it like this if you prefer it this way

Reply:
thanks! I changed that